tests/test_checkout.py

```diff
@@ -510,3 +510,24 @@ def assertNewLineFollows(self, finish_bar):
 
     def assertAnyEndsWith(self, update_bars, name):
         self.assertTrue(any(ub for ub in update_bars if ub.endswith(name)))
+
+
+class TestCheckoutTargetRecursiveShouldNotRemoveOtherUsedFiles(TestDvc):
+    def test(self):
+        storage = self.mkdtemp()
+
+        ret = main(["remote", "add", "-d", "storage", storage])
+        self.assertEqual(0, ret)
+
+        ret = main(["add", self.DATA_DIR, self.FOO, self.BAR])
+        self.assertEqual(0, ret)
+
+        ret = main(["push"])
+        self.assertEqual(0, ret)
+
+        # There is no direct way to checkout recursive
```

Comment:
You could call `self.dvc.checkout(self.DATA_DIR, recursive=True)` through API though and won't need this remote stuff here. Also, you could even simply add `dvc checkout -R` support along the way :) Your choice.

Reply:
Ill go with support, sooner or later, someone will need that.